image_data_sar/sched.py

```diff
@@ -0,0 +1,264 @@
+                try:
+                    aca.make_report()
+                except Exception:
+                    err = traceback.format_exc()
+                    aca.add_message('critical', text=f'Running make_report() failed:\n{err}')
+
+            if aca.roll_info:
+                aca.make_roll_options_report()
+
+            aca.context['text_pre'] = aca.get_text_pre()
+            aca.context['call_args'] = aca.get_call_args_pre()
+
+    # noinspection PyDictCreation
+    if make_html:
+        # from . import __version__
+        __version__ = "0.1"
+
+        # Create new context or else use a copy of the supplied dict
+        if context is None:
+            context = {}
+        else:
+            context = context.copy()
+
+        context['load_name'] = load_name.upper()
+        context['proseco_version'] = proseco.__version__
+        context['sparkles_version'] = __version__
+        context['chandra_aca_version'] = chandra_aca.__version__
+        context['acas'] = acars
+        context['summary_text'] = get_summary_text(acars)
+
+        # Special case when running a set of roll options for one obsid
+        is_roll_report = all(aca.is_roll_option for aca in acars)
+
+        label_frame = 'ACA' if aca.is_ER else 'Target'
+        context['id_label'] = f'{label_frame} roll' if is_roll_report else 'Obsid'
+
+        template_file = SPARKLES_DIR / 'index_template_preview.html'
+        template = Template(open(template_file, 'r').read())
+        out_html = template.render(context)
+
+        out_filename = report_dir / 'index.html'
+        if loud:
+            print(f'Writing output review file {out_filename}')
+        with open(out_filename, 'w') as fh:
+            fh.write(out_html)
+
+        if open_html:
+            import webbrowser
+            out_url = f'file://{out_filename.absolute()}'
+            print(f'Open URL in browser: {out_url}')
+            webbrowser.open(out_url)
+
+
+def sub_in_stars(aca):
+
+    # Sort the candidates by mag
+    aca.guides.cand_guides.sort('mag')
+    args = aca.call_args.copy()
+
+    # Sub in the faintest and stop if it still has no criticals
+    for i, n in enumerate(range(len(aca.guides.cand_guides) - 1, len(aca.guides) - 1, -1)):
+        star1 = aca.guides.cand_guides['id'][n]
+        star2 = aca.guides.cand_guides['id'][n - 1]
+        print(f"trying candidates {star1} {star2}")
+        args['include_ids_guide'] = [star1, star2]
+        naca = proseco.get_aca_catalog(**args)
+        nacar = naca.get_review_table()
+        nacar.run_aca_review()
+        if len(nacar.messages == 'critical') == 0:
+            break
+    return naca, star1, star2
+
+
+def do(load_name):
+    acas = pickle.load(gzip.open(load_name))
+    aca_arr = [acas[cat] for cat in acas]
+    aca_arr = sorted(aca_arr, key=lambda k: k.meta['date'])
+
+    # Do we want to parse the ms to check the last one?  Otherwise no duration.
+    candidates = []
+    for idx, aca in enumerate(aca_arr[0:-1]):
```

Comment:
It might make sense for the ACA reviewer to just manually check the schedule summary to see if the last dwell is a long ER.